server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -198,21 +194,14 @@ private String getSymptom(
 
     private List<HealthIndicatorImpact> getImpacts(
         Set<String> indicesWithBlock,
-        Set<String> indicesOnRedNodes,
-        Set<String> indicesOnYellowNodes,
-        Set<String> nodesWithBlockedIndices,
         Set<String> redDataNodes,
         Set<String> yellowDataNodes,
-        Set<String> redMasterNodes,
-        Set<String> yellowMasterNodes,
-        Set<String> redNonDataNonMasterNodes,
-        Set<String> yellowNonDataNonMasterNodes
+        Set<String> nodesReportingRed,
+        Set<String> nodesReportingYellow,
+        ClusterState clusterState
     ) {
         List<HealthIndicatorImpact> impacts = new ArrayList<>();
-        if (indicesWithBlock.isEmpty() == false
-            || indicesOnRedNodes.isEmpty() == false
-            || nodesWithBlockedIndices.isEmpty() == false
-            || redDataNodes.isEmpty() == false) {
+        if (indicesWithBlock.isEmpty() == false || redDataNodes.isEmpty() == false) {
```

Comment:
Follow up PR material: 

The impact now says "Cannot insert or update documents in the affected indices."

I wonder if we'd like to list a few indices (truncate to 10 like we do in `shards_availability`)

What do you think?

Reply:
Hm, we can indeed. It might not be very helpful if a user has ilm set-up or data streams but for others it will probably be very helpful.

Reply:
Added it, it was actually very useful to highlight some edge cases that the message didn't make as much sense as I thought.